enhancements/microshift/kubernetes-for-device-edge.md

```diff
@@ -0,0 +1,515 @@
+standalone OpenShift, allowing us to maintain binary compatibility.
+The versions of the embedded components are selected by looking at an
+OpenShift release image, finding the git SHAs for the direct
+dependencies, resolving the secondary dependencies, and vendoring the
+results into the MicroShift git repository. That process needs to be
+as automatic as possible, so we want to avoid carrying any
+MicroShift-specific patches that must be applied after a rebase. The
+version of the containerized components are taken from the same
+release image and the image references are built into the MicroShift
+binary.
+
+A great deal of effort has gone into making OpenShift 4 self-managing,
+especially when it comes to multi-node deployment and upgrades. We do
+not want to reinvent that work, nor do we want to recreate the
+OpenShift 3 model using an outside orchestration tool.  A single-node
+control plane becomes a single point of failure when worker nodes are
+attached, which eliminates the high availability benefits of running
+workloads on Kubernetes. Those statements taken together lead us to
+conclude that MicroShift will only support deployments with 1 node
+acting as both control plane and worker.
+
+Because MicroShift does not reuse the system operators from OpenShift,
+it will not be configured through API resources in the way that
+OpenShift is. Instead, we will use a new configuration file for the
+control plane (`/etc/microshift/config.yaml`) with [limited
+configuration
+options](https://github.com/openshift/microshift/blob/main/pkg/config/config.go). The
+node-side configuration will be done via the standard kubelet
+configuration file.
+
+Other operating system services will be configured outside of
+MicroShift using their standard configuration files. There is no
+machine-config-operator.
+
+Because MicroShift will be treated as a single application, we will
+not expose separate health checks or implement per-component restart
+features. If any embedded component is unhealthy, the application will
+be treated as unhealthy and be restarted.
+
+### Workflow Description
+
+#### Deploying
+
+Application deployers will use RHEL for Edge tools such as Image
+Builder to construct full system images containing RHEL, MicroShift,
+agents, custom operating system configuration, data prerequisites, and
+optionally application images. Those system images will be written to
+the target devices either individually (via the RHEL installation
+process) for development/testing or via manufacturer imaging processes
+for large-scale production deployments. RHEL for Edge uses
+ostree-based images, and an [example
+workflow](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md)
+for constructing a basic image is described in the MicroShift
+documentation. We are not creating a separate installer for
+MicroShift.
+
+#### Upgrading
+
+Because the operating system is based on rpm-ostree, upgrades are
```

Comment:
s/upgrades/ OS upgrades/

Reply:
It's not just the OS. If you want a new MicroShift version on the same old OS version, you still build a new system image and upgrade using ostree.